src/main/java/cs505/grad1/sensoragg/Grad1Component.java

```diff
@@ -0,0 +1,171 @@
+package cs505.grad1.sensoragg;
+
+import java.util.HashMap;
+import java.util.Map;
+import org.iot.raspberry.grovepi.GrovePi;
+import cs505.grad1.sensoragg.mock.*;
+
+/**
+ * Grad1Component is a public class used to create a list of sensors and start
+ * the factory process.
+ *
+ * @version 1.0.0
+ * @author Colin Kelly
+ * @author Ben Marshalkowski
+ * @author Kathleen McKay
+ * @author Phil Murray
+ */
+public class Grad1Component {
+
+	private Map<Integer, SensorType> sensors = new HashMap();
+	private AbstractAggregatedDataFactory factory;
+
+	/**
+	 * Port offset to avoid HashMap collisions between analog and digital ports
+	 */
+	static final int digitalOffset = 32;
+
+	/**
+	 * Port offset to avoid HashMap collisions between a sensors which share a port
+	 */
+	static final int secondarySensorOffset = 16;
+
+	public Grad1Component(MockGrovePi mock) {
+			this.factory = new MockFactory(mock);
+	};
+
+	public Grad1Component(GrovePi grovepi) {
+			this.factory = new AggregatedDataFactory(grovepi);
+	}
+
+	public Grad1Component(Builder builder) {
+			this.sensors = builder.sensors;
+			if (builder.grovePi != null)
+				this.factory = new AggregatedDataFactory(builder.grovePi);
+			else
+				this.factory = new MockFactory(builder.mock);
+	}
+
+	/**
+	 * Inner class for BUILDER PATTERN. Rather than instantiating a {@link Grad1Component} object using a
+	 * constructor and then adding on the various
+	 * sensors using the user can "build" the
+	 * Grad1Component with the help of this inner class. For example:
+	 * Grad1Component grad1 = new Grad1Component.Builder().sensor(1, SensorType.LIGHT).sensor(2, SensorType.RANGER).grovePi(new GrovePi4J).build();
+	 */
+	public static class Builder {
```

Comment:
Builder is a nice addition for common usage scenario and explanation of it in Javadoc